src/clib/cool1d_multi_g-cpp.C

```diff
@@ -0,0 +1,1740 @@
+                + gasH2Oice  [i-1] * (tgas[i-1] - grain_temperatures.data[OnlyGrainSpLUT::H2O_ice_dust]  [i-1])
+                ) * d(i-1,idx_range.jp1-1,idx_range.kp1-1) * rhoH[i-1];
+            }
+          }
+
+
+        }
+
+        edot[i-1] = edot[i-1] + Ldst[i-1];
+
+      }
+    }
+
+  }
+
+  // Compute continuum opacity
+
+  if ( my_chemistry->use_primordial_continuum_opacity == 1 )  {
+
+    for (i = idx_range.i_start + 1; i<=(idx_range.i_end + 1); i++) {
+      if ( itmask[i-1] != MASK_FALSE )  {
+
+        // ! primordial continuum opacity !!
+         FORTRAN_NAME(interpolate_2d_g)(
+          &logrho[i-1], &logT[i-1], my_rates->alphap.props.dimension, my_rates->alphap.props.parameters[0], &my_rates->alphap.props.parameter_spacing[0],
+          my_rates->alphap.props.parameters[1], &my_rates->alphap.props.parameter_spacing[1], &my_rates->alphap.props.data_size,
+          my_rates->alphap.data, &log_a);
+        alpha[i-1] = std::pow(1.e1,log_a);
+      }
+    }
+
+  } else {
+    for (i = idx_range.i_start + 1; i<=(idx_range.i_end + 1); i++) {
+      if ( itmask[i-1] != MASK_FALSE )  {
+        alpha[i-1] = 0.f;
+      }
+    }
+
+  }
+
+  // Add dust opacity.
+  // if (idspecies .eq. 0), dust opacity is overestimated at Td > 50 K
+  // We better not include dust opacity.
+  if ((anydust != MASK_FALSE) && (my_chemistry->dust_species > 0))  {
+    for (i = idx_range.i_start + 1; i<=(idx_range.i_end + 1); i++) {
+      if ( itmask_metal[i-1] != MASK_FALSE )  {
+
+        if (my_chemistry->use_multiple_dust_temperatures == 0)  {
+
+          alphad[i-1] = kptot[i-1];
+
+        } else {
+
+          if (my_chemistry->dust_species > 0)  {
+            alphad[i-1] = kpMgSiO3  [i-1]
+                      + kpAC      [i-1];
+          }
+          if (my_chemistry->dust_species > 1)  {
+            alphad[i-1] = alphad[i-1]
+                      + kpSiM     [i-1]
+                      + kpFeM     [i-1]
+                      + kpMg2SiO4 [i-1]
+                      + kpFe3O4   [i-1]
+                      + kpSiO2D   [i-1]
+                      + kpMgO     [i-1]
+                      + kpFeS     [i-1]
+                      + kpAl2O3   [i-1];
+          }
+          if (my_chemistry->dust_species > 2)  {
+            alphad[i-1] = alphad[i-1]
+                      + kpreforg  [i-1]
+                      + kpvolorg  [i-1]
+                      + kpH2Oice  [i-1];
+          }
+        }
+
+        alpha[i-1] = alpha[i-1] + alphad[i-1] * d(i-1,idx_range.jp1-1,idx_range.kp1-1)*dom*mh_local_var;
+
+      }
+    }
+
+  }
+
+  for (i = idx_range.i_start + 1; i<=(idx_range.i_end + 1); i++) {
+    if ( itmask[i-1] != MASK_FALSE )  {
+      tau_con[i-1] = alpha[i-1] * lshield_con[i-1];
+    }
+  }
+
+  // --- Compute (external) radiative heating terms ---
+  // Photoionization heating
+
+  if (my_chemistry->primordial_chemistry > 0)  {
+
+    if (my_chemistry->self_shielding_method == 0)  // no shielding
+    {
```

Comment:
Could be deferred to a future PR

```suggestion
    if (my_chemistry->self_shielding_method == 0) { // no shielding
```